server/src/main/java/org/elasticsearch/gateway/AsyncShardFetch.java

```diff
@@ -56,6 +56,7 @@
      */
     public interface Lister<NodesResponse extends BaseNodesResponse<NodeResponse>, NodeResponse extends BaseNodeResponse> {
         void list(ShardId shardId, @Nullable String customDataPath, DiscoveryNode[] nodes, ActionListener<NodesResponse> listener);
+        void flush();
```

Comment:
Rather than introducing this method to the lister (and the corresponding flag passed in to `fetchData`) could we have the allocator directly indicate the end of an allocation round which triggers the flush.